DEV-4/src/abstractClasses/Counter.java

```diff
@@ -0,0 +1,25 @@
+package abstractClasses;
+
+/**
+ * Abstract class which provides his methods to _Counter classes
+ * it's generic because outputer receives object of this class
+ * so it could be every type what we want
+ */
+public abstract class Counter<T> {
+
+  /**
+   * parent method return null
+   * can be used as cap
+   * overrides in child classes
+   */
+  public T getValue() {
+    return null;
```

Comment:
what? why null? to have one chance to have NullPointerException? If you don't no specific imp[implementation then make it abstract